### PR TITLE
Add Stack Exchange Presence

### PR DESCRIPTION
**Pull request by avikrao**
Creating a presence for Stack Exchange and all of its subdomains and all the domains superuser.com, askubuntu.com, and serverfault.com, all of which are sites under Stack Exchange's network. Did NOT add support for Stack Overflow, as it already has a standalone presence.
image
image
image

### CODE REVIEW

File: websites/S/Stack Exchange/presence.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
+const presence = new Presence({
+    clientId: "919817726195814431"
+  }),
+  startTimestamp = Math.floor(Date.now() / 1000);
```
Comment (EncryptedDev) [inline websites/S/Stack Exchange/presence.ts:4]:
```suggestion
  browsingTimestamp = Math.floor(Date.now() / 1000);
```

File: websites/S/Stack Exchange/presence.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    presenceData.largeImageKey = "askubuntu";
+    presenceData.details = "Ask Ubuntu Meta";
+  } else if (hostname === "superuser.com") {
+    presenceData.largeImageKey = "superuser";
+    presenceData.details = "Super User";
+  } else if (hostname === "meta.superuser.com") {
+    presenceData.largeImageKey = "superuser";
+    presenceData.details = "Super User Meta";
+  } else {
+    const subStack = document.querySelector("meta[property='og:site_name']"),
```
Comment (EncryptedDev) [inline websites/S/Stack Exchange/presence.ts:33]: Variable is only used once

File: websites/S/Stack Exchange/presence.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        "superuser.com",
+        "meta.superuser.com",
+        "askubuntu.com",
+        "meta.askubuntu.com"
+      ].includes(hostname)
+    )
+      presenceData.state = "Main Page";
+    else presenceData.details = "Main Page";
+  } else if (pathname.includes("/questions")) {
+    const titleElem = document.querySelector(".question-hyperlink");
```
Comment (EncryptedDev) [inline websites/S/Stack Exchange/presence.ts:59]: Variable is only used once

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/S/Stack Exchange/dist/metadata.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{
+    "$schema": "https://schemas.premid.app/metadata/1.5",
+    "author": {
+      "name": "AcidEpicice",
+      "id": "249992970235936768"
+    },
+    "url": ["stackexchange.com", "serverfault.com", "askubuntu.com", "superuser.com"],
+    "regExp": "(.*\\.)*stackexchange\\.com|superuser\\.com|askubuntu\\.com|serverfault\\.com",
```
Comment (ImRodry) [inline websites/S/Stack Exchange/dist/metadata.json:8]: this seems rather redundant, unless you want to match all of those domains with anything before them, in which case the regex is incorrect
```suggestion
    "regExp": "(.*\\.)*stackexchange\\.com",
```

File: websites/S/Stack Exchange/dist/metadata.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    "color": "#1E5397",
+    "tags": [
+      "help",
+      "community",
+      "discussion",
+      "forum",
+      "question",
+      "answer"
+    ],
+    "category": "other"
```
Comment (ImRodry) [inline websites/S/Stack Exchange/dist/metadata.json:25]: move above tags pls

**Review by ImRodry — CHANGES_REQUESTED** [review]

File: websites/S/Stack Exchange/dist/metadata.json (hunk 4 above)
Comment (AvikRao) [reply websites/S/Stack Exchange/dist/metadata.json:8]: Sorry, could you elaborate what the issue with the regex is? If you mean that I'm incorrectly catching subdomains, I _do_ want to catch all subdomains of the listed sites

Comment (ImRodry) [reply websites/S/Stack Exchange/dist/metadata.json:8]: This regex will match `ggerhgyuierhbgre.stackexchange.com` but not `ergergerger.serverfault.com`, which is what you want

Comment (AvikRao) [reply websites/S/Stack Exchange/dist/metadata.json:8]: Looks like it does, actually?
image

Comment (ImRodry) [reply websites/S/Stack Exchange/dist/metadata.json:8]: It shouldn’t then, here’s the correct one (do test this though)
```suggestion
    "regExp": "$(.*\\.)*(stackexchange\\.com|superuser\\.com|askubuntu\\.com|serverfault\\.com)^",
```

**Review by Slowlife01 — COMMENTED** [review]
There's already a presence for AskUbuntu

**Comment by AvikRao** [comment]
> There's already a presence for AskUbuntu

removed askubuntu

**Review by Slowlife01 — APPROVED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**